llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -5585,7 +5619,7 @@ void BoUpSLP::reorderTopToBottom() {
 
   // Reorder the graph nodes according to their vectorization factor.
   for (unsigned VF = VectorizableTree.front()->getVectorFactor(); VF > 1;
-       VF = bit_ceil(VF) / 2) {
+       VF = (VF & ~1U) - 2) {
```

Comment:
What is this trying to do?  & -1 should just be VF shouldn't it?

Reply:
If VF is odd (for 3, 7, etc. number of elements), want to try next even. I reworked it to `VF += 2 - (VF & 1)` to jump over even VFs